src/transformers/models/vitpose/image_processing_vitpose.py

```diff
@@ -0,0 +1,566 @@
+                Representation for Human Pose Estimation (CVPR 2020).
+            kernel (int):
+                aussian kernel size (K) for modulation, which should match the heatmap gaussian sigma when training.
+                K=17 for sigma=3 and k=11 for sigma=2.
+            valid_radius_factor (float):
+                The radius factor of the positive area in classification heatmap for UDP.
+            use_udp (bool):
+                Use unbiased data processing.
+            target_type (str):
+                'GaussianHeatmap' or 'CombinedTarget'.
+                GaussianHeatmap: Classification target with gaussian distribution.
+                CombinedTarget: The combination of classification target
+                (response map) and regression target (offset map).
+                Paper ref: Huang et al. The Devil is in the Details: Delving into
+                Unbiased Data Processing for Human Pose Estimation (CVPR 2020).
+
+        Returns:
+            tuple: A tuple containing keypoint predictions and scores.
+
+            - preds (np.ndarray[batch_size, num_keypoints, 2]):
+                Predicted keypoint location in images.
+            - maxvals (np.ndarray[batch_size, num_keypoints, 1]):
+                Scores (confidence) of the keypoints.
+        """
+        # Avoid being affected
+        heatmaps = heatmaps.numpy().copy()
+
+        batch_size, num_keypoints, height, width = heatmaps.shape
+
+        preds, maxvals = _get_max_preds(heatmaps)
+
+        preds = post_dark_udp(preds, heatmaps, kernel=kernel)
+
+        # Transform back to the image
+        for i in range(batch_size):
+            preds[i] = transform_preds(preds[i], center[i], scale[i], [width, height], use_udp=use_udp)
+
+        return preds, maxvals
+
+    def post_process_pose_estimation(self, outputs, boxes, kernel_size=11, use_udp=True):
+        """
+        Transform the heatmaps into keypoint predictions and transform them back to the image.
+
+        Args:
+            outputs (torch.Tensor):
+                Model outputs.
+            boxes (torch.Tensor of shape [batch_size, 4]):
+                Bounding boxes.
+            kernel_size (`int`, *optional*, defaults to 11):
+                Gaussian kernel size (K) for modulation.
+            use_udp (`bool`, *optional*, defaults to `False`):
+                Whether to use unbiased data processing.
+        """
+
+        # First compute centers and scales for each bounding box
+        batch_size = len(outputs.heatmaps)
+        centers = np.zeros((batch_size, 2), dtype=np.float32)
+        scales = np.zeros((batch_size, 2), dtype=np.float32)
+        for i in range(batch_size):
+            width, height = self.size["width"], self.size["height"]
+            center, scale = box_to_center_and_scale(boxes[i], image_width=width, image_height=height)
+            centers[i, :] = center
+            scales[i, :] = scale
```

Comment:
Model outputs should be processed as torch tensors

Reply:
Do you mean I should replace all numpy calls by PyTorch?

Reply:
Yes. I realise this isn't entirely consistent across the image processors, e.g. Beit converts to numpy, but many post_process methods return tensors, and it makes more sense to keep the outputs consistent with the pre-processed inputs

Reply:
This still needs to be addressed

Reply:
@amyeroberts @NielsRogge pose_result is already converted to tensor type. Only the intermediate value is used in numpy. Should I change these intermediate values to numpy also?

Reply:
Unresolved

Reply:
> Should I change these intermediate values to numpy also?

All the returned values should be in the same format (numpy or torch). They would ideally all be returned as torch tensors, but all numpy is acceptable